### PR TITLE
Mongoose: add “names” selector version of Document#populate

### PR DESCRIPTION
**Pull request by sindrenm**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.
  - Not present.

**Changing an existing definition**

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://mongoosejs.com/docs/populate.html
- [ ] Increase the version number in the header if appropriate.
  - Not sure if this is appropriate. I looked at the chanelog from 4.7.0 through 4.7.3,
    where I know it's present, and didn't find it included there.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.
  - Not a substantial change.

### CODE REVIEW

**Comment by sindrenm** [comment]
The `tsc` command failed from before as well with the same errors as it does now. It's not related to this change.

File: mongoose/index.d.ts [hunk 1]
```diff
@@ -994,10 +994,12 @@ declare module "mongoose" {
      * call execPopulate(). Passing the same path a second time will overwrite
      * the previous path options. See Model.populate() for explaination of options.
      * @param path The path to populate or an options object
+     * @param names The properties to fetch from the populated document
      * @param callback When passed, population is invoked
      */
     populate(callback: (err: any, res: this) => void): this;
     populate(path: string, callback?: (err: any, res: this) => void): this;
+    populate(path: string, names: string, callback?: (err: any, res: this) => void): this;
```
Comment (simonxca) [inline mongoose/index.d.ts]: @sindrenm Thanks!

File: mongoose/index.d.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // Type definitions for Mongoose 4.7.0
 // Project: http://mongoosejs.com/
-// Definitions by: simonxca <https://github.com/simonxca/>, horiuchi <https://github.com/horiuchi/>
+// Definitions by: simonxca <https://github.com/simonxca/>, horiuchi <https://github.com/horiuchi/>, sindrenm <https://github.com/sindrenm>
```
Comment (simonxca) [inline mongoose/index.d.ts]: @sindrenm this means you'll be notified of future PRs and will help review them. If you're not interested in doing that, you should remove this.

**Comment by simonxca** [comment]
@sindrenm yea, travis error not you're fault. Can you add `// TypeScript Version: 2.1` to the header like https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/lodash/index.d.ts#L5? This should fix it.

**Comment by sindrenm** [comment]
Sure, I'll do that. Thanks, @simonxca!

**Comment by sindrenm** [comment]
You're right, that seems to have done the trick. :smile:

File: mongoose/index.d.ts (hunk 2 above)
Comment (sindrenm) [inline mongoose/index.d.ts]: Thank you for making this such an easy thing to add in the first place. :+1:

**Comment by sindrenm** [comment]
Thanks for merging! :heart:

**Pull request merged**